### ISSUE TITLE
Missing parens in `cmp_null`'s example

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#16281]
### Description

The example in `cmp_null` gives `x == ptr::null`, while the valid code of this is `x == ptr::null()`. Also, in its description it only mentions `ptr::null`, while it actually checks for both `ptr::null` and `ptr::null_mut`

### Version

```text
rustc 1.94.0-nightly (f52090008 2025-12-10)
binary: rustc
commit-hash: f5209000832c9d3bc29c91f4daef4ca9f28dc797
commit-date: 2025-12-10
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.5
```

**Comment by profetia** [comment]
This lint also handles macros wrongly, for this code:
```rust
fn wrongly_handles_macros() {
    use std::ptr;

    struct Container {
        value: *const i32,
    }
    let x = Container { value: ptr::null() };

    macro_rules! dot_value {
        ($obj:expr) => {
            $obj.value
        };
    }

    if dot_value!(x) == ptr::null() {
        todo!()
    }
}
```
